tests/OptionsTest.php

```diff
@@ -14,6 +14,27 @@ final class OptionsTest extends TestCase
 {
     use ExpectDeprecationTrait;
 
+    /**
+     * @var int
+     */
+    private $errorReportingOnSetup;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        // There are tests changing the error reporting level in this class so we store the current value before running the tests
+        $this->errorReportingOnSetup = error_reporting();
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        // There are tests changing the error reporting level in this class so we restore it to the value before running the tests
+        error_reporting($this->errorReportingOnSetup);
```

Comment:
Order of calls should be reversed, we need to intervene before other teardown operations are triggered.

Also, comments are probably superfluous here, it's pretty clear what we're doing.

Reply:
Ah yeah makes sense, didn't think about that, switched it around!